Library/Homebrew/test/test_formula_lock.rb

```diff
@@ -19,4 +19,16 @@ def test_locking_file_with_existing_lock_raises_error
   def test_locking_existing_lock_suceeds
     assert_nothing_raised { @lock.lock }
   end
+
+  def test_locking_file_in_nonexistent_cache_directory
+    @tmp_lockdir = FormulaLock::LOCKDIR
+    FormulaLock.send(:remove_const, :LOCKDIR)
+    FormulaLock.const_set(:LOCKDIR, @tmp_lockdir + "nonsuch/")
+    @nonsuch_dir = FormulaLock.new("bar")
+    assert_nothing_raised { @nonsuch_dir.lock }
+    assert_nothing_raised { @nonsuch_dir.unlock }
+    FormulaLock::LOCKDIR.children.each(&:unlink)
+    FormulaLock.send(:remove_const, :LOCKDIR)
+    FormulaLock.const_set(:LOCKDIR, @tmp_lockdir)
+  end
```

Comment:
You might want to use `ensure` to, well, _ensure_ stuff is correctly reset if the test fails:

```rb
def test_locking_file_in_nonexistent_cache_directory
  # the test goes here
ensure
  # the code that resets things goes here
end
```

See [`test_integration_cmds.rb`](https://github.com/Homebrew/homebrew/blob/8a26671c16096d8b6b8bc51f76a3b21643cbd4af/Library/Homebrew/test/test_integration_cmds.rb#L327-L339) for some examples.

The issue right now is that if e.g. the first `assert_nothing_raised` fails, the test stops here and the rest of the code is not executed.